tests/contrib/aiopg/py37/test.py

```diff
@@ -0,0 +1,48 @@
+# 3p
+import aiopg
+
+# project
+from ddtrace.contrib.aiopg.patch import patch, unpatch
+from ddtrace import Pin
+
+# testing
+from tests.contrib.config import POSTGRES_CONFIG
+from tests.contrib.asyncio.utils import AsyncioTestCase, mark_asyncio
+
+
+TEST_PORT = str(POSTGRES_CONFIG['port'])
+
+
+class AiopgTestCase(AsyncioTestCase):
+    # default service
+    TEST_SERVICE = 'postgres'
+
+    def setUp(self):
+        super().setUp()
+        self._conn = None
+        patch()
+
+    def tearDown(self):
+        super().tearDown()
+        if self._conn and not self._conn.closed:
+            self._conn.close()
+
+        unpatch()
+
+    async def _get_conn_and_tracer(self):
+        conn = self._conn = await aiopg.connect(**POSTGRES_CONFIG)
+        Pin.get_from(conn).clone(tracer=self.tracer).onto(conn)
+
+        return conn, self.tracer
+
+    @mark_asyncio
+    async def test_async_generator(self):
+        conn, _ = await self._get_conn_and_tracer()
+        cursor = await conn.cursor()
+        q = 'select \'foobarblah\''
+        await cursor.execute(q)
+        rows = []
+        async for row in cursor:
+            rows.append(row)
+
+        assert rows == [('foobarblah',)]
```

Comment:
We should probably assert on any spans created as well here.

Something like this should be enough: https://github.com/DataDog/dd-trace-py/blob/9adad57d6021fe0df035248e342936c622bfabe1/tests/contrib/aiopg/py35/test.py#L54-L57

Reply:
it is more of being 100% sure that all code path/workflows result in the correct behavior, which is 1) not crashing and 2) producing the correct spans we expect.

You don't to have to go too crazy with asserting on spans like other test cases that assert on name, service, resource, meta, etc etc, a simple assertion like the above to ensure we have created the desired number of spans and it was a `postgres.query` span like we expected is good enough.

Reply:
Done :)

BTW, I've gone ahead and rebased, so your merge from master wasn't needed, I hope it's what you expected !